### ISSUE TITLE
Ignore unknown fields in representation of externally tagged enum

### ISSUE THREAD

**Issue by whimboo** [serde-rs/serde#1377]
Hi,

I have a problem with an untagged enum, which in same cases can get keys which are legacy and by default not used anymore. During deserialization with Serde I just want that those are ignored. Right now it raises an error like "data did not match any variant of untagged enum".

Here is a simplified playground example:
https://play.rust-lang.org/?gist=ec22336dd31436f130b9b433f8394d61&version=stable&mode=debug&edition=2015

Basically I don't want to create a helper enum with all the possible legacy keys, but simply ignore any of them. Similar to what we have for `struct` enabled by default. Is something like that possible?

**Comment by dtolnay** [comment]
Your playground link does not include an untagged enum. Is this the wrong link?

**Comment by whimboo** [comment]
Sorry, but that issue was wrongly phrased. It's indeed not an untagged enum. I have updated the summary. The main question btw. is still the same.

**Comment by dtolnay** [comment]
I would recommend writing the Deserialize impl yourself, inside of which you can implement your custom logic for selecting a variant. [Playground link](https://play.rust-lang.org/?gist=cb29faf2d16967483e369276f19638cd&version=stable&mode=debug&edition=2015)

```rust
impl<'de> Deserialize<'de> for Origin {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        let value = Value::deserialize(deserializer)?;
        if value == "Pointer" {
            Ok(Origin::Pointer)
        } else if value == "Viewport" {
            Ok(Origin::Viewport)
        } else if let Some(web_element) = value.get("element-6066-11e4-a52e-4f735466cecf") {
            WebElement::deserialize(web_element)
                .map(Origin::Element)
                .map_err(de::Error::custom)
        } else {
            Err(de::Error::custom("invalid value of enum Origin"))
        }
    }
}
```

**Comment by whimboo** [comment]
This works like a charm. Thank you so much.

**Comment by ul** [comment]
Is it possible to make this behaviour default (or opt-in with attribute) in Serde without the need to write custom deserializer? It looks like quite common case (e.g. https://github.com/gluon-lang/lsp-types/issues/101) and will be consistent with ignoring extra fields in plain structs.

**Comment by Marwes** [comment]
@ul This is for an externally tagged enum, not an untagged.
